README.md

```diff
@@ -53,6 +53,12 @@ See more examples in the `./samples` folder.
 You can use the `HTTPS_PROXY` or `https_proxy` environment variable for proxy configuration.
 Read more about it [here](https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/).
 
+### Exponential backoff configuration
+
+You can configure exponential that will be applied to all outgoing requests from the SDK. You may configure this through a `PlatformUrlOptions` object.
+
+Outgoing requests will retry when a `429` status code is returned from the platform. By default, the will wait retry a maximum of 50 times, waiting 5 seconds between attempts, with a time multiple of 2 (5 seconds for first attempt, 10 for second, 20 for third, etc).
```

Comment:
I think we'll need much more meat than this.

* Why do we need retry ?
* What does exponential mean exactly ?
* Any details you think would be helpful to give about the internals of the mechanism ?
* Any article we can link from docs.coveo.com ? I don't think any is available, but confirm with the guys in charge of the project.
* Adding the code sample done above here instead, with each option, and their default value.

```typescript
new CatalogSource('my_api_key', 'my_coveo_organization_id', {
     // The maximum amount of retry. Defaults to 50.
     // After this maximum as been reached, the SDK will throw an error
     // etc etc...
     maxRetries: 50,
     // [...]
     retryAfter:  5000,
     timeMultiple: 2
});
```

Said differently, as yourself if you would realistically be able to figure out how to configure the SDK by just reading this doc and transforming that to a code snippet.

Reply:
I've rewritten this section of the README, it should be much more clear now.

Moreover, there isn't anything on the docs site for the SDKs and they instead just point to the github pages, so I strongly suspect this will be the only documentation for this. I think this should be good enough, but let me know if you think it's worthwhile getting an API writer involved to review these docs ✅